sources/cfsignal/src/cloudformation.rs

```diff
@@ -0,0 +1,55 @@
+use crate::error::{self, Result};
+use imdsclient::ImdsClient;
+use log::info;
+use rusoto_cloudformation::{CloudFormation, CloudFormationClient, SignalResourceInput};
+use rusoto_core::Region;
+use snafu::{OptionExt, ResultExt};
+use std::str::FromStr;
+
+pub async fn signal_resource(
+    stack_name: String,
+    logical_resource_id: String,
+    status: String,
+) -> Result<()> {
+    info!("Connecting to IMDS");
+    let mut client = ImdsClient::new();
+    let instance_id = get_instance_id(&mut client).await.unwrap();
+    let region = get_region(&mut client).await.unwrap();
```

Comment:
`unwrap` would cause the program to panic when they occur, let's try avoiding that.

We can just float these errors up to the caller if they do occur.
```suggestion
    let instance_id = get_instance_id(&mut client).await?;
    let region = get_region(&mut client).await?;
```